python/test/test_forward_all.py

```diff
@@ -0,0 +1,309 @@
+# Copyright (c) 2017 Sony Corporation. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from six.moves import range
+import pytest
+import numpy as np
+import nnabla as nn
+import nnabla.functions as F
+import nnabla.parametric_functions as PF
+
+
+def initialize_grad(parameters):
+    for param in parameters.values():
+        param.grad.zero()
+
+@pytest.mark.parametrize("seed", [313])
+def test_graph_logreg(seed):
+    rng = np.random.RandomState(seed)
+    x = nn.Variable([2, 3, 4], need_grad=True)
+    w1 = nn.Variable([12, 5], need_grad=True)
+    w2 = nn.Variable([12, 5], need_grad=True)
+    b1 = nn.Variable([5], need_grad=True)
+    b2 = nn.Variable([5], need_grad=True)
+    t = nn.Variable([2, 1])
+    x.d = rng.randn(*x.shape)
+    w1.d = rng.randn(*w1.shape)
+    w2.d = rng.randn(*w2.shape)
+    b1.d = rng.randn(*b1.shape)
+    b2.d = rng.randn(*b2.shape)
+    t.d = rng.randint(0, 5, size=t.shape)
+
+    nn.set_default_context(nn.Context())
+
+    # Forwardprop by definintion
+    z1 = F.affine(x, w1, b1, 1)
+    z2 = F.affine(x, w2, b2, 1)
+    l1 = F.softmax_cross_entropy(z1, t, 1)
+    L1 = F.mean(l1)
+    l2 = F.softmax_cross_entropy(z2, t, 1)
+    L2 = F.mean(l2)
+    nn.forward_all([L1, L2])
+
+    # Backprop for z1
+    # Diff should be initialized since they are always accumulated
+    x.g = 0
+    w1.g = 0
+    b1.g = 0
+    L1.backward(clear_buffer=True)
```

Comment:
@TE-AkioHayakawa 
The test has passed with the latest changes. But, I don't know why this `clear_buffer=True` makes this test passed.